WebdriverIO.md

```diff
@@ -0,0 +1,561 @@
+    * [`core/tests/webdriverio_desktop`](#coretestswebdriverio_desktop)
+    * [`core/tests/webdriverio_mobile`](#coretestswebdriverio_mobile)
+  * [Utilities](#utilities)
+    * [`core/tests/webdriverio_utils`](#coretestswebdriverio_utils)
+    * [`extensions/**/webdriverio.js`](#extensionswebdriverios)
+* [Run E2E tests](#run-e2e-tests)
+  * [Set chromedriver version](#set-chromedriver-version)
+  * [Configure test sharding](#configure-test-sharding)
+  * [Run a single end-to-end test](#run-a-single-end-to-end-test)
+  * [Run end-to-end tests in production mode](#run-end-to-end-tests-in-production-mode)
+* [Write E2E tests](#write-e2e-tests)
+  * [Where to add the tests](#where-to-add-the-tests)
+    * [Interactions](#interactions)
+    * [Existing suite](#existing-suite)
+    * [New suite](#new-suite)
+  * [Writing the tests](#writing-the-tests)
+  * [Writing utilities](#writing-utilities)
+    * [Selecting elements](#selecting-elements)
+    * [Non-Angular pages](#non-angular-pages)
+  * [Writing robust tests](#writing-robust-tests)
+    * [Flakiness](#flakiness)
+    * [Independence](#independence)
+  * [Checking for flakiness](#checking-for-flakiness)
+  * [Codeowner Checks](#codeowner-checks)
+  * [Important Tips](#important-tips)
+* [Metrics](#metrics)
+* [Reference](#reference)
+  * [Forms and objects](#forms-and-objects)
+    * [Rich Text](#rich-text)
+  * [Async-Await Tips](#async-await-tips)
+    * [Good Patterns](#good-patterns)
+    * [Anti-Patterns](#anti-patterns)
+
+## Layout of the E2E test files
+
+E2E test logic is divided between two kinds of files: suite files and utility files. Utility files provide functions for interacting with pages, for example by clicking buttons or checking that the expected text is visible. Suite files define the E2E tests using calls to the utility files.
+
+Suppose you wanted to write an E2E test that changes a user's profile picture and then checks that the change was successful. Your utility file might define `setProfilePicture()` and `checkProfilePicture` functions. Then your suite file would first call `setProfilePicture()` and then call `checkProfilePicture()`.
+
+### Suite files
+
+Note that "suite files" are also known as "test files."
+
+#### `core/tests/webdriverio`
+
+This directory contains test suites which were applicable to both desktop and mobile interfaces. However, we don't run the mobile tests anymore. Certain operations were possible only on one or the other interface. To distinguish between the interfaces, we use the boolean, `browser.isMobile` defined in `onPrepare` of the webdriverio configuration file. Even though we don't run the mobile tests anymore, you might see some legacy code that uses this boolean.
+
+#### `core/tests/webdriverio_desktop`
+
+This directory houses all test suites which are exclusive to desktop interfaces. This generally includes core creator components like the rich-text editor.
+
+#### `core/tests/webdriverio_mobile`
```

Comment:
Are you planning to migrate these even though we don't use them anymore?

Reply:
I think we are adding the mobile tests in one GSoC project (Making Contributor Dashboard UI Responsive). Please see [this](https://github.com/oppia/oppia/pull/15536)